pkg/apis/apps/validation/validation.go

```diff
@@ -94,6 +96,32 @@ func ValidatePersistentVolumeClaimRetentionPolicy(policy *apps.StatefulSetPersis
 	return allErrs
 }
 
+func volumesToAddForTemplates(spec *apps.StatefulSetSpec) []api.Volume {
```

Comment:
See below comment, I think this needs to return volumes for all the VolumeClaimTemplates, and returning a map[name]volume will make it easier to use in the loop below

```suggestion
func volumesToAddForTemplates(spec *apps.StatefulSetSpec) map[string]api.Volume {
```